dds/DCPS/RTPS/Spdp.cpp

```diff
@@ -1012,248 +1149,234 @@ Spdp::handle_handshake_message(const DDS::Security::ParticipantStatelessMessage&
 
   DiscoveredParticipant& dp = iter->second;
 
-  switch (dp.auth_state_) {
-  case DCPS::AUTH_STATE_AUTHENTICATED:
-    dp.auth_state_ = DCPS::AUTH_STATE_UNKNOWN;
-    attempt_authentication(src_participant, dp);
-    break;
-  case DCPS::AUTH_STATE_HANDSHAKE:
-    break;
-  default:
-    if (DCPS::security_debug.auth_warn) {
-      ACE_DEBUG((LM_WARNING, ACE_TEXT("(%P|%t) {auth_warn} Spdp::handle_handshake_message() - ")
-                 ACE_TEXT("Dropping handshake message from %C, auth_state_ %d not expected\n"),
-                 OPENDDS_STRING(DCPS::GuidConverter(src_participant)).c_str(),
-                 dp.auth_state_));
-    }
-    return;
+  if (DCPS::security_debug.auth_debug) {
+    ACE_DEBUG((LM_DEBUG, "(%P|%t) {auth_debug} DEBUG: Spdp::handle_handshake_message for %C auth_state=%d handshake_state=%d\n", OPENDDS_STRING(DCPS::GuidConverter(src_participant)).c_str(), dp.auth_state_, dp.handshake_state_));
   }
 
+  // We have received a handshake message from the remote which means
+  // we don't need to send the auth req.
+  dp.have_auth_req_msg_ = false;
+
   switch (dp.handshake_state_) {
-  case DCPS::HANDSHAKE_STATE_EXPECTING_REPLY_OR_FINAL:
-    if (msg.related_message_identity.source_guid != guid_) {
-      if (DCPS::security_debug.auth_warn) {
-        ACE_DEBUG((LM_WARNING,
-                   ACE_TEXT("(%P|%t) {auth_warn} Spdp::handle_handshake_message() - ")
-                   ACE_TEXT("Dropping handshake message from %C, incorrect related message identity source guid\n"),
-                   OPENDDS_STRING(DCPS::GuidConverter(src_participant)).c_str()));
-      }
-      return;
-    }
+  case DCPS::HANDSHAKE_STATE_BEGIN_HANDSHAKE_REQUEST:
+  case DCPS::HANDSHAKE_STATE_WAITING_FOR_TOKEN:
+  case DCPS::HANDSHAKE_STATE_DONE: {
+    ACE_ERROR((LM_ERROR,
+               ACE_TEXT("(%P|%t) {auth_warn} Spdp::handle_handshake_message() - ")
```

Comment:
```suggestion
               ACE_TEXT("(%P|%t) Spdp::handle_handshake_message() - ")
```

Reply:
Or make it actually make it conditional on the `auth_warn` logging tag.